File: cpp/program30.cpp
// program25.c

#include <iostream>
using namespace std;

class Number
{
    public:
        int iNo;

    Number(int X)
    {
        iNo = X;
    }

    int SumFactors()
    {
        int iCnt = 0;
        int iSum = 0;

        if(iNo < 0)
        {
            iNo = -iNo;
        }

        for(iCnt = 1; iCnt <= (iNo/2); iCnt++)
        {
            if((iNo % iCnt) == 0)
            {
                iSum = iSum + iCnt;
            }
        }
        return iSum;
    }

};

int main()
{
    int iValue = 0;
    int iRet = 0;

    cout<<"Enter Number"<<"\n";
    cin>>iValue;

    Number nobj(iValue);
    iRet = nobj.SumFactors();

    cout<<"Summation of the Factors is : "<<iRet<<"\n";

    return 0;
}

File: java/program30.java
import java.util.*;

class Number
{
    public int iNo;

    public Number(int no)
    {
        iNo = no;
    }
    
    public int SumFactors()
    {
        int iSum = 0,iCnt = 0;

        for(iCnt = 1; iCnt <= (iNo/2); iCnt++)
        {
            if((iNo%iCnt) == 0)
            {
                iSum = iSum + iCnt;
            }
        }
        return iSum;
    }
}

class program30
{
    public static void main(String arg[])
    {
        int iValue = 0,iRet = 0;
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number ");
        iValue = sobj.nextInt();

        Number obj = new Number(iValue);
        iRet = obj.SumFactors();

        System.out.println("Sum of the Factors are : "+iRet);
    }
}